### PR TITLE
Verify file checksum generator name

### PR DESCRIPTION
**Pull request by ajkr**
Previously we only had a debug assertion to check the right generator was being used for verification. However a user hit a problem in production where their factory was creating the wrong generator for some files, leading to checksum mismatches. It would have been easier to debug if we verified in optimized builds that the generator with the proper name is used. This PR adds such verification.

### CODE REVIEW

File: file/file_util.cc [hunk 1]
```diff
@@ -151,14 +151,21 @@ IOStatus GenerateOneFileChecksum(
         requested_checksum_func_name +
         " from checksum factory: " + checksum_factory->Name();
     return IOStatus::InvalidArgument(msg);
+  } else {
+    // For backward compatibility, requested_checksum_func_name can be empty.
```
Comment (zhichao-cao) [inline file/file_util.cc:155]: Maybe add a sentence of comments on line 128 to indicate if requested function name does not match the one from factory, IOStatus::InvalidArgument is returned

**Review by zhichao-cao — APPROVED** [review]
LGTM, thanks for the fix!

File: file/file_util.cc (hunk 1 above)
Comment (ajkr) [reply file/file_util.cc]: Good idea, thanks. Done.

File: file/file_util.cc [hunk 2]
```diff
@@ -151,14 +153,22 @@ IOStatus GenerateOneFileChecksum(
         requested_checksum_func_name +
         " from checksum factory: " + checksum_factory->Name();
     return IOStatus::InvalidArgument(msg);
+  } else {
+    // For backward compatibility and use in file ingestion clients where there
```
Comment (ajkr) [inline file/file_util.cc:157]: I also updated this comment a bit since it looks like empty string is passed during file ingestion , which appears unrelated to compatibility.

**Review by ajkr — COMMENTED** [review]
Thanks for the review!
